packages-native/carousel/src/Carousel.tsx

```diff
@@ -0,0 +1,166 @@
+
+        if (paginationOverflow) {
+            return (
+                <View style={pagination.container} {...a11yProps}>
+                    <Text style={pagination.text}>
+                        {activeSlide + 1}/{contentLength}
+                    </Text>
+                </View>
+            );
+        }
+
+        return (
+            <Pagination
+                dotsLength={contentLength}
+                activeDotIndex={activeSlide}
+                containerStyle={pagination.container}
+                dotContainerStyle={pagination.dotContainerStyle}
+                dotColor={pagination.dotStyle?.color}
+                dotStyle={pagination.dotStyle}
+                inactiveDotStyle={pagination.inactiveDotStyle}
+                inactiveDotColor={pagination.inactiveDotStyle?.color}
+                inactiveDotOpacity={pagination.inactiveDotStyle?.opacity}
+                inactiveDotScale={pagination.inactiveDotStyle?.scale}
+                carouselRef={carouselRef}
+                tappableDots
+                {...a11yProps}
+            />
+        );
+    }, [activeSlide, carouselRef, props.contentSource, props.showPagination]);
+
+    const onLayout = (event: LayoutChangeEvent) => {
+        let viewHeight = event.nativeEvent.layout.height;
+        const viewWidth = event.nativeEvent.layout.width;
+
+        let itemWidth = 0;
+        let itemHeight = 0;
+
+        if (layoutSpecificStyle.slideItem) {
+            const { width: slideItemWidth, height: slideItemHeight } = layoutSpecificStyle.slideItem;
+            // We calculate the actual number value in order to
+            // allow users to set width and height as percentage since lib only accepts numbers
+
+            if (typeof slideItemWidth === "string" && slideItemWidth.includes("%")) {
+                const percentage = +slideItemWidth.replace("%", "");
+                itemWidth = (viewWidth * percentage) / 100;
+            } else {
+                itemWidth = Number(slideItemWidth);
+            }
+            if (typeof slideItemHeight === "string" && slideItemHeight.includes("%")) {
+                const percentage = +slideItemHeight.replace("%", "");
+                itemHeight = (viewWidth * percentage) / 100;
+            } else {
+                itemHeight = Number(slideItemHeight);
+            }
+
+            if (styles.container?.height === undefined && itemHeight > 0) {
+                viewHeight = itemHeight;
+            }
+        }
+
+        setSliderDimensions({
+            slider: { width: viewWidth, height: viewHeight },
+            slide: { width: itemWidth, height: itemHeight }
+        });
+    };
+
+    return (
+        <View style={styles.container} onLayout={onLayout}>
+            {loading ? (
+                <ActivityIndicator color={layoutSpecificStyle.indicator!.color} size="large" />
+            ) : (
+                /* library ignores width/height if its vertical/horizontal */
```

Comment:
Comments?

Reply:
I comment this here because it took me some time to find this magic behaviour in the readme of the library. I think it will be useful 1-2 months later when someone else look back at the code.